hyperliquid/exchange.py

```diff
@@ -97,9 +119,10 @@ def cancel(self, coin: str, oid: int) -> Any:
                 "type": "cancel",
                 "cancels": [
                     {
-                        "asset": asset,
-                        "oid": oid,
+                        "asset": cancel["coin"],  # TODO: are asset and coin the same thing?
```

Comment:
asset and coin aren't the same thing. There's a one to one mapping between assets and coins, assets are numbers the exchange uses, whereas coins are the generally accepted tickers for them. For example "ETH" is a coin who's asset is 1. You can convert between then by doing self.coin_to_asset[cancel["coin"]] here as you did above